index.js

```diff
@@ -0,0 +1,83 @@
+// dependencies
+var async = require('async');
+var AWS = require('aws-sdk');
+var util = require('util');
+
+var mailparse = require('./parse.js');
+// get reference to S3 client 
+var s3 = new AWS.S3();
+
+
+
+exports.handler = function(event, context, callback) {
+    // Read options from the event.
+    var eventBody = event.Records[0].body;
+    var srcBucket = JSON.parse(eventBody).Records[0].s3.bucket.name;
+    // Object key may have spaces or unicode non-ASCII characters.
+    var srcKey    = decodeURIComponent(JSON.parse(eventBody).Records[0].s3.object.key.replace(/\+/g, " "));  
+    var dstBucket = srcBucket + "resized";
```

Comment:
Take the destination bucket as confit as well, don’t assume anything about the configuration.